extensions/qute/qute-reference-guide.adoc

```diff
@@ -0,0 +1,673 @@
+
+----
+{#if item.name is 'sword'}
+  It's a sword!
+{#else if item.name is 'shield'}
+  It's a shield!
+{#else}
+  Item is nor a sword nor a shield.
+{/if}
+----
+
+===== Loop Section
+
+The loop section makes it possible to iterate over an instance of `Iterable`, `Map` 's entry set and `Stream`.
+It has two flavors.
+The first one is using `each` name alias.
+
+----
+{#each items}
+ {it.name} <1>
+{/each}
+----
+<1> `it` is an implicit alias. `name` is resolved against the current iteration element.
+
+The other form is using `for` name alias and can specify the alias used to reference the iteration element:
+----
+{#for item in items}
+  {item.name}
+{/for}
+----
+
+It's also possible to access the iteration metadata inside the loop:
+----
+{#each items}
+ {count}. {it.name} <1>
+{/each}
+----
+<1> `count` represents one-based index. Metadata also include zero-based `index`, `hasNext`, `odd`, `even`.
+
+===== If Section
+
+A basic control flow section.
+The simplest possible version accepts a single parameter and renders the contet if it's evaluated to `true` (or `Boolean.TRUE`).
+
+----
+{#if item.active}
+ This item is active.
+{/if}
+----
+
+You can also use the following operators:
+
+|===
+|Operator |Aliases 
+
+|equals 
+|`eq`, `==`, `is` 
+
+|not equals 
+|`ne`, `!=`
+
+|greater than 
+|`gt`, `>`
+
+|greater equals
+|`ge`, `>=`
+
+|less than 
+|`lt`, `<`
+
+|less equals 
+|`le`, `<=`
+
+|===
+
+----
+{#if item.age > 10}
+ This item is very old.
+{/if}
+----
+
+NOTE: Multiple conditions are not supported.
+
+You can add any number of "else" blocks:
+
+----
+{#if item.age > 10}
+ This item is very old.
+{#else if item.age > 5}
+ This item is quite old.
+{#else if item.age > 2}
+ This item is old.
+{#else}
+ This item is not old at all!
+{/if}
+----
+
+===== With Section
+
+This section can be used to set the current context object.
+This could be useful to simplify the template structure.
+
+----
+{#with item.parent}
+ {name}  <1>
+{/with}
+----
+<1> The name will be resolved against the `item.parent`.
+
+It's also possible to specify an alias for the context object:
+
+----
+{#with item.parent as myParent}
+ {myParent.name}
+{/with}
+----
+
+[[include_helper]]
+===== Include/Insert Sections
+
+These sections can be used to include another template and possibly override some parts of the template (template inheritance).
+
+.Template "base"
+[source,html]
+----
+<html>
+<head>
+<meta charset="UTF-8">
+<title>{#insert title}Default Title{/}</title> <1>
+</head>
+<body>
+    {#insert body}No body!{/} <2>
+</body>
+</html>
+----
+<1> `insert` sections are used to specify parts that could be overriden by a template that includes the given template.
+<2> An `insert` section may define the default content that is rendered if not overriden.
+
+.Template "detail"
+[source,html]
+----
+{#include base} <1>
+    {#title}My Title{/title} <2>
+    {#body}
+    <div>
+        My body.
+    </div>
```

Comment:
Is it normal you don't end the body block?

It probably is considering the note below but it feels a bit weird.

Reply:
Yes, end tag is optional for section blocks. The start tag of an additional section block is usually more like a separator:

```
{#if foo}
...
{#else} -> the same situation
...
{/if} 
```